core/src/main/java/com/linecorp/armeria/server/InternalServiceRequestContext.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import io.netty.buffer.ByteBufAllocator;
+
+/**
+ * An interface defining internal-only methods of a {@link ServiceRequestContext}. These methods should be used
+ * with care, generally only in generic services and protocol implementations, not user business logic.
+ */
+public interface InternalServiceRequestContext extends ServiceRequestContext {
```

Comment:
Could be `InternalRequestContext extends RequestContext`, given that we can apply the same optimization for outgoing requests?